### DISCUSSION TITLE
rxjs 7.1.0 subscribe() error parameter empty

### DISCUSSION THREAD

**Discussion by DFDevPriv** [ReactiveX/rxjs#6518] [Q&A]
Hi all,
This is my first time starting a discussion on github and I hope you can help me, as I have no idea
at the moment how to fix this problem.
I'm facing a problem since upgrading rxjs 6.3.3 to 7.1.0 with the error handling of subscribe(). Since the upgrade
the error parameter is undefined and for me it looks like this could be a bug, as there is no problem with the next parameter.
My example code looks like this:
```
    onSubmit() {
        this.loading = true;
        this.backendService.login(this.username, this.password, true).subscribe({
            next: (data) => {
                console.log(data);
                // Do some more stuff here
            },
            error: (error) => {
                console.log("Error occured");
                Object.keys(error).forEach(function(key,index) {
                    console.log(key); // key: the name of the object key
                    // index: the ordinal position of the key within the object 
                });
                this.loading = false;
                if("status" in error && error["status"]==401){
                    this.modalService.setModalInfo({
                        "title": "Login Failed",
                        "subtitle": 'User credentials incorrect'
                    });
                } else {
                    this.modalService.setModalError({
                        "body": 'Error during login. ' + error['error']['error']
                    }); 
                }
            }
        })
```
The 401 response from the webservice looks like this:
```
{
  "error": "Authentication Failed"
}
```
and the error message is:
ERROR TypeError: Cannot read property 'error' of undefined

If more information is needed, just let me know.

Thanks all for your help in advance!

Regards,
Daniel

**Comment by voliva** [comment]
I'm not sure what could be wrong in here - Where is that `TypeError` being raised?

The only thing I can think that's happening is this line:

```ts
{
"body": 'Error during login. ' + error['error']['error']
}
```

What's the value of the argument `error`? If it doesn't look like:

```ts
error = {
  error: {
     error: "error"
  }
}
```

it could throw the exception you're showing here.

**Reply by DFDevPriv** [reply]
The error is already thrown on the line where I tried to get the attributes (for debugging purposes) of the error object:
`Object.keys(error).forEach(function(key,index) {
    console.log(key); // key: the name of the object key
    // index: the ordinal position of the key within the object 
});`
And there the error object is undefined, that is why I'm wondering if there is a bug in the subscribe method as the error data which was sent with the 401 code by the webservice isn't available as it was with rxjs version 6.3.3
I have the same problem in all parts of the application where the subscribe runs into the error handling.
